packages/mermaid/src/utils/imperativeState.ts

```diff
@@ -0,0 +1,37 @@
+export const createImperativeState = <S extends Record<string, unknown>>(init: () => S) => {
+  const state = init();
+
+  return {
+    get records() {
+      return state;
+    },
+    reset: () => {
+      Object.keys(state).forEach((key) => {
+        delete state[key];
+      });
+      Object.entries(init()).forEach(([key, value]: [keyof S, any]) => {
+        state[key] = value;
+      });
```

Comment:
Would making state a mutable variable and doing `state = init();` here cause any issues?

Reply:
I want to keep reference equality after reset, but it is not a big deal